aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPluginTest.kt

```diff
@@ -256,4 +264,94 @@ class RealAWSCognitoAuthPluginTest {
+        val onError = mockk<Consumer<AuthException>>(relaxed = true)
+        val listenLatch = CountDownLatch(1)
+        val currentAuthState = mockk<AuthState> {
+            every { authNState } returns AuthenticationState.SignedIn(mockk())
+        }
+        every { authStateMachine.getCurrentState(captureLambda()) } answers {
+            lambda<(AuthState) -> Unit>().invoke(currentAuthState)
+        }
+
+        val credential = AmplifyCredential(
+            CognitoUserPoolTokens("idToken", "accessToken", "refreshToken", 120L),
+            null,
+            null
+        )
+
+        val eventSlot = CapturingSlot<(CredentialStoreState) -> Unit>()
+        every { credentialStoreStateMachine.listen(capture(eventSlot), any()) } answers {
+            eventSlot.captured.invoke(CredentialStoreState.Success(credential))
+            UUID.randomUUID()
+        }
+
+        val userAttributes = listOf(
+            AttributeType.invoke {
+                name = "email"
+                value = "email"
+            },
+            AttributeType.invoke {
+                name = "nickname"
+                value = "nickname"
+            }
+        )
+
+        val expectedResult = buildList {
+            userAttributes.mapTo(this) {
+                AuthUserAttribute(
+                    AuthUserAttributeKey.custom(it.name),
+                    it.value
+                )
+            }
+        }
+
+        coEvery {
+            authService.cognitoIdentityProviderClient?.getUser(any<GetUserRequest>())
+        } returns GetUserResponse.invoke {
+            this.userAttributes = userAttributes
+        }
+
+        every {
+            onSuccess.accept(expectedResult.toMutableList())
+        } answers {
+            listenLatch.countDown()
+        }
+
+        // WHEN
+        plugin.fetchUserAttributes(onSuccess, onError)
+
+        assertTrue { listenLatch.await(5, TimeUnit.SECONDS) }
+        coVerify(exactly = 0) { onError.accept(any()) }
+    }
+
+    @Test
+    fun `fetch user attributes fails when not in SignedIn state`() {
+        // GIVEN
+        val onSuccess = mockk<Consumer<MutableList<AuthUserAttribute>>>(relaxed = true)
+        val onError = mockk<Consumer<AuthException>>(relaxed = true)
+        val listenLatch = CountDownLatch(1)
+        val currentAuthState = mockk<AuthState> {
+            every { authNState } returns AuthenticationState.NotConfigured()
+        }
+        every { authStateMachine.getCurrentState(captureLambda()) } answers {
+            lambda<(AuthState) -> Unit>().invoke(currentAuthState)
+        }
+
+        every {
+            onError.accept(AuthException.InvalidStateException())
+        } answers {
+            listenLatch.countDown()
+        }
+
+        // WHEN
+        plugin.fetchUserAttributes(onSuccess, onError)
+
+        assertTrue { listenLatch.await(5, TimeUnit.SECONDS) }
+        verify(exactly = 0) { onSuccess.accept(any()) }
```

Comment:
Shouldn't we verify if the error was called instead as per test title? (And verify success in above test cases)

Reply:
If I understand correctly, what you are saying is that in line 346, where you are mocking the behavior of onError and execute countdown, you assume that the onError was called because latch counted down? Wouldn't it still make sense to have
verify {onError.accept(...)} still so that test still passes if we add latch.countdown on any other line?

Reply:
Yes I agree and added the change